site/content/entry/citra-progress-report-2020-q2.md

```diff
@@ -0,0 +1,699 @@
+
+More modern consoles often have high-level FS APIs, which means that the game requests the OS to read files. While this means more work for the initial implementation, it also means that LayeredFS can be added with ease simply by replacing the data to be returned to the game. For the 3DS, however, things are different. The FS service only provides basic functions to read the RomFS as a huge binary chunk - and the games need to parse the filesystem themselves.
+
+[Luma3DS](https://github.com/AuroraWright/Luma3DS)'s LayeredFS works by hooking the SDK code the games include to parse the RomFS, but [zhaowenlan1779](https://github.com/zhaowenlan1779) decided to take a harder and more robust approach. With help from 3dbrew documents, he wrote code to rebuild the RomFS metadata and assign fake data offsets to files. From the game's perspective, it would seem as if the RomFS has been completely rebuilt, but we are not actually doing that much work.
+
+(Picture of a game with fancy mods)
+
+## Disk Shader Cache ([#4923](https://github.com/citra-emu/citra/pull/4923)) by [jroweboy](https://github.com/jroweboy)
+
+Do you like it if you game just hanges for a few seconds repeatedly? No? Then this feature is something for you.
+
+Stuttery gameplay is a real issue in some games on Citra like Luigi's Mansion: Dark Moon.
+This game is developed with a very modern approach to GPU hardware and makes extensive use of the 3DS GPU for all of its effects.
+
+A shader is a small program that runs on your GPU. 
+Whenever the game tells Citra to run a new shader, Citra translates this shader, so that your GPU can understand it. Then, it gets uploaded to your GPU.
+This entire process can take a bit of time, so Citra has to pause emulation until the shader is ready, causing stutter whenever the game is drawing something for the first time.
+
+With this change, Citra will now instead preload all of the shaders you've encountered during gameplay, allowing you to play through those same spots again without any stutter.
+
+(Gif of loading stutter)
+(Gif of same place after cached)
+Comparison of what shader stutter looks like before and after its cached.
+
+## Video Dumper ([#4602](https://github.com/citra-emu/citra/pull/4602), [#5083](https://github.com/citra-emu/citra/pull/5083)) by [zhaowenlan1779](https://github.com/zhaowenlan1779)
+
+Some of you may be wondering - since it is so easy to use OBS to record gameplay, what is the point of a builtin video dumper? Well, while OBS is cool, there are people out there who wish to share their gameplay at consistent speeds without slowdowns and stutters. TAS communitys like [TASVideos](http://tasvideos.org/) also required complying emulators to provide a way to 'render' gameplay. That was why [zhaowenlan1779](https://github.com/zhaowenlan1779) implemented a video dumper into Citra around Aug 2019.
```

Comment:
Hmm, thinking again, I'd like it to be something like
`Some of you may be wondering: since it is so easy ... gameplay, what is the point of ...?`

Reply:
> 
> 
> Hmm, thinking again, I'd like it to be something like
> `Some of you may be wondering: since it is so easy ... gameplay, what is the point of ...?`

Mhm... So how about this:
`Some of you may be wondering: Since it is so easy to use OBS to record gameplay... What ist the point of a built-in video dumper?`